lib/Widgets/SettingsPage/AbstractSimpleSettingsPage.vala

```diff
@@ -0,0 +1,159 @@
+/*
+* Copyright (c) 2017 elementary LLC. (http://launchpad.net/switchboard-plug-security-privacy)
+*
+* This program is free software; you can redistribute it and/or
+* modify it under the terms of the GNU General Public
+* License as published by the Free Software Foundation; either
+* version 2 of the License, or (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+* General Public License for more details.
+*
+* You should have received a copy of the GNU General Public
+* License along with this program; if not, write to the
+* Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+* Boston, MA 02110-1301 USA
+*
+*/
+
+/**
+ * SimpleSettingsPage is a widget divided into three sections: a predefined header,
+ * a content area, and an action area.
+ */
+
+public abstract class Granite.SimpleSettingsPage : Granite.SettingsPage {
+    private Gtk.Image header_icon;
+    private Gtk.Label description_label;
+    private Gtk.Label title_label;
+    private string _description;
+    private string _icon_name;
+    private string _title;
+
+    /**
+     * A #Gtk.ButtonBox used as the action area for #this
+     */
+    public Gtk.ButtonBox action_area { get; construct; }
+
+    /**
+     * A #Gtk.Grid used as the content area for #this
+     */
+    public Gtk.Grid content_area { get; construct; }
+
+    /**
+     * A #Gtk.Switch that appears in the header area when #this.activable is #true
+     */
+    public Gtk.Switch status_switch;
+
+    /**
+     * Creates a #Gtk.Switch #status_switch in the header of #this
+     */
+    public bool activatable { get; construct; }
```

Comment:
I think this `activatable` should be still controlable, so that the switch could be shown / hidden dynamically.